RecoMET/METFilters/plugins/HFNoisyHitsFilter.cc

```diff
@@ -0,0 +1,148 @@
+#include "FWCore/Framework/interface/global/EDFilter.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "Geometry/CaloTopology/interface/CaloTopology.h"
+#include "Geometry/CaloEventSetup/interface/CaloTopologyRecord.h"
+
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+
+#include "DataFormats/HcalRecHit/interface/HFRecHit.h"
+#include "DataFormats/HcalRecHit/interface/HcalRecHitCollections.h"
+#include "DataFormats/METReco/interface/HcalPhase1FlagLabels.h"
+//
+// class declaration
+//
+
+class HFNoisyHitsFilter : public edm::global::EDFilter<> {
+public:
+  explicit HFNoisyHitsFilter(const edm::ParameterSet&);
+  ~HFNoisyHitsFilter() override {}
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  bool filter(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+  std::vector<int> getNoiseBits() const;
+  const edm::EDGetTokenT<HFRecHitCollection> hfhits_token_;
+  const double rechitPtThreshold_;
+  const std::vector<std::string> listOfNoises_;
+  const bool taggingMode_;
+  const bool debug_;
+};
+
+//
+// constructors and destructor
+//
+HFNoisyHitsFilter::HFNoisyHitsFilter(const edm::ParameterSet& iConfig)
+    : hfhits_token_(consumes<HFRecHitCollection>(iConfig.getParameter<edm::InputTag>("hfrechits"))),
+      rechitPtThreshold_(iConfig.getParameter<double>("rechitPtThreshold")),
+      listOfNoises_(iConfig.getParameter<std::vector<std::string>>("listOfNoises")),
+      taggingMode_(iConfig.getParameter<bool>("taggingMode")),
+      debug_(iConfig.getParameter<bool>("debug")) {
+  produces<bool>();
+}
+
+//
+// member functions
+//
+
+// ------------ method called on each new Event  ------------
+bool HFNoisyHitsFilter::filter(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
+  using namespace edm;
+  bool pass = true;
+
+  // Calo Geometry - needed for computing E_t
+  ESHandle<CaloGeometry> pG;
+  iSetup.get<CaloGeometryRecord>().get(pG);
+  const CaloGeometry* geo = pG.product();
+
+  edm::Handle<HFRecHitCollection> theHFhits;
+  iEvent.getByToken(hfhits_token_, theHFhits);
+
+  std::vector<int> noiseBits = getNoiseBits();
+
+  //Loop over the HF rechits. If one of them has Et>X and fires one the noise bits, declare the event as bad
+  for (HFRecHitCollection::const_iterator hfhit = theHFhits->begin(); hfhit != theHFhits->end(); hfhit++) {
+    float ene = hfhit->energy();
+    float et = 0;
+    // compute transverse energy
+    const GlobalPoint& poshf = geo->getPosition(hfhit->detid());
+    float pf = poshf.perp() / poshf.mag();
+    et = ene * pf;
+    if (et < rechitPtThreshold_)
+      continue;
+    int hitFlags = hfhit->flags();
+    for (unsigned int i = 0; i < noiseBits.size(); i++) {
+      if (noiseBits[i] < 0)
+        continue;
+      if (((hitFlags >> noiseBits[i]) & 1) > 0) {
```

Comment:
```suggestion
      if ((hitFlags >> noiseBits[i]) & 1) {
```
this is apparently enough